rules/static_framework_generator.bzl

```diff
@@ -0,0 +1,214 @@
+
+def _declare_framework(ctx, target):
+    framework_name = _get_existing_framework_name(target)
+    return _declare_directory(ctx, framework_name, framework_name + ".framework")
+
+def _make_frameworks(ctx):
+    frameworks = []
+    for target in ctx.attr.targets:
+        framework = _declare_framework(ctx, target)
+        resource_bundles = _declare_resource_bundles(ctx, target)
+
+        existing_framework_files = target.files.to_list()
+        existing_framework_path = _get_existing_framework_path(target)
+
+        resource_paths = " ".join([bundle.path for bundle in resource_bundles])
+        ctx.actions.run_shell(
+            outputs = [framework],
+            inputs = existing_framework_files + resource_bundles,
+            command = 'rsync -a "$1/" "$3"; for resource in $2; do rsync -a "$resource" "$3/Resources"; done',
+            arguments = [existing_framework_path, resource_paths, framework.path],
+        )
+        frameworks.append(framework)
+    return frameworks
+
+def _make_destination(ctx):
+    if ctx.attr.destination_relative_to_package:
+        return ctx.label.package + "/" + ctx.attr.destination_relative_to_package
+    else:
+        return ctx.attr.destination_relative_to_workspace
+
+def _make_installer(ctx, subpath, files, destination):
+    installer = ctx.actions.declare_file(subpath + "/installer.sh")
+    ctx.actions.expand_template(
+        template = ctx.file._installer_template,
+        output = installer,
+        substitutions = {
+            "$(files)": " ".join(files),
+            "$(destination)": destination,
+        },
+        is_executable = True,
+    )
+    return installer
+
+def _static_framework_generator_impl(ctx):
+    frameworks = _make_frameworks(ctx)
+    destination = _make_destination(ctx)
+    installer = _make_installer(ctx, ctx.label.name, [framework.path for framework in frameworks], destination)
+    return [
+        DefaultInfo(
+            executable = installer,
+            runfiles = ctx.runfiles(files = frameworks),
+        ),
+    ]
+
+# Making below rule public in order to properly generate its documentation until the
+# https://github.com/bazelbuild/stardoc/issues/27 issue is resolved
+static_framework_generator_test = rule(
+    implementation = _static_framework_generator_impl,
+    doc = """\
+This generator moves the frameworks artifacts produced by Bazel to a location out of the cache.
+It is intended to be used as a tool to ease compatibility with other build systems.
+
+For example, it would be possible to integrate with cocoapods as follows:
+1- Build targets with Bazel
+2- Use this rule to generate frameworks for them
+3- Generate podspec files containing the generated frameworks as vendored_frameworks, so they can be 
+   integrated in a cocoapods setup, substituting their source code counterparts (useful to cut build time)
+""",
+    attrs = {
+        "targets": attr.label_list(
```

Comment:
Does this this cause a reinstall per target? Any reason why we shouldn't have a per target installer?

Reply:
Here I chose an array to allow for both options: you can declare the rule once per target (getting one installer per target), or you can declare one rule for multiple targets (one installer for all)